src/modules/battery.rs

```diff
@@ -105,16 +105,16 @@ pub fn get_batteries() -> Result<Vec<BatteryInfo>, ModuleError> {
             // Err(e) => return Err(ModuleError::new("Battery", format!("Can't read from {} - {}", path, e))),
             Err(_) => continue,
         };
-        let id: String = match path.file_name() {
-            Some(r) => match r.to_str() {
-                Some(r) => r.strip_prefix("BAT").unwrap_or(r).to_string(),
+        let model_name: String = match util::file_read(&path.join("model_name")) {
+            Ok(r) => r.trim().to_string(),
+            Err(_) => match path.file_name() {
+                Some(r) => r.to_str().unwrap_or("Unknown").to_string(),
                 None => continue,
```

Comment:
Why not return the supply name as before instead?

Reply:
This is the fallback if we fail to acquire the model name. A string is better than just an index.

In addition, it's `macsmc-battery` for asahi, not `BATx`

Reply:
Ah nvm I missed it somehow, ignore me lol